js/views/modals/Settings/Advanced.js

```diff
@@ -0,0 +1,78 @@
+import app from '../../../app';
+import loadTemplate from '../../../utils/loadTemplate';
+import baseVw from '../../baseVw';
+import $ from 'jquery';
+
+export default class extends baseVw {
+  constructor(options = {}) {
+    super({
+      className: 'settingsAdvanced',
+      ...options,
+    });
+
+    this.settings = app.settings.clone();
+
+    this.listenTo(this.settings, 'sync', () => app.settings.set(this.settings.toJSON()));
+  }
+
+  getFormData(subset = this.$formFields) {
+    return super.getFormData(subset);
+  }
+
+  saveServer() {
+    const formData = this.getFormData();
+
+    this.settings.set(formData);
+
+    const save = this.settings.save();
+
+    this.trigger('saving');
+
+    if (!save) {
+      // client side validation failed
+      this.trigger('saveComplete', true);
+    } else {
+      this.trigger('savingToServer');
+
+      save
+        .done(() => {
+          this.trigger('saveComplete');
+        })
+        .fail((...args) => {
+          this.trigger('saveComplete', false, true,
+            args[0] && args[0].responseJSON && args[0].responseJSON.reason || '');
+        });
+    }
+
+    this.render();
+    const $firstErr = this.$('.errorList:first');
+    if ($firstErr.length) $firstErr[0].scrollIntoViewIfNeeded();
+  }
+
+  saveLocal() {
+    const localData = this.getFormData(this.$localFields);
+    app.localSettings.set(localData);
```

Comment:
Instead of working on the global app.localSettings, could you please make a clone of it in the constructor and work off of that? Also, please bind a sync handler to the cloned localSettings and update the global one there. Basically, give localSettings the same treatment as app.settings is getting.

The reason is that we don't want invalid data (data that failed client validation) making it into the global localSettings.